scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,209 @@
+);
+
+# Allowed certificate thumbprints
+$global:ValidThumbprints = @( `
+    "B9EAA034C821C159B05D3521BCF7FEB796EBD6FF", `
+    "98ED99A67886D020C564923B7DF25E9AC019DF26", `
+    "5EAD300DC7E4D637948ECB0ED829A072BD152E17", `
+    "67B1757863E3EFF760EA9EBB02849AF07D3A8080", `
+    "9DC17888B5CFAD98B3CB35C1994E96227F061675", `
+    "62009AAABDAE749FD47D19150958329BF6FF4B34", `
+    "3BDA323E552DB1FDE5F4FBEE75D6D5B2B187EEDC" `
+)
+
+# Allow-list for unsigned binaries
+$global:UnsignedAllowList = @{
+    # "PackageId" = relative paths
+    "Microsoft.TestPlatform" = @( `
+        ".\tools\net451\Common7\IDE\Extensions\TestPlatform\Newtonsoft.Json.dll" `
+    )
+
+    "Microsoft.TestPlatform.CLI" = @( `
+        ".\contentFiles\any\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\contentFiles\any\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+    
+    "Microsoft.TestPlatform.Portable" = @( `
+        ".\tools\net451\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+}
+
+
+$global:Depth = 0;
+
+function Get-RelativePath($path, $relativeTo) {
+    Push-Location
+    Set-Location $relativeTo
+    $path = Resolve-Path $path -Relative
+    Pop-Location
+    
+    return $path
+}
+
+function Push-Depth {
+    $global:Depth = $global:Depth + 1
+}
+
+function Pop-Depth {
+    $global:Depth = $global:Depth - 1
+
+    if($global:Depth -lt 0) {
+        $global:Depth = 0;
+    }
+}
+
+function Clear-Line($message = "") {
+    if(-not $CIBuild) {
+        $x = [System.Console]::get_CursorLeft()
+        Write-Host "`r$(' ' * $x)$message`r" -NoNewline
+    }
+}
+
+function Update-Operation($operation, $message, [switch]$NewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Yellow -NoNewline
+    Write-Host $message -NoNewline
+
+    if ($newline -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Submit-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Green -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Undo-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Red -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Expand-NugetPackages {
+    $foldersToProcess = [System.Collections.Generic.HashSet[string]]@()
+
+    $nupgks = Get-ChildItem *.nupkg
+
+    Push-Depth
+    $nupgks | ForEach-Object {
+        $package = $_.FullName

```

Comment:
If you do Get-Item here you can use `$package.Directory`, `$package.BaseName` etc. instead of the long winded IO.Path methods.

Reply:
You marked this as resolved but don't actually do it.